core/src/main/scala/scalaz/LazyEitherT.scala

```diff
@@ -105,7 +105,7 @@ object LazyEitherT extends LazyEitherTInstances {
     u1: Unapply[Functor, FAB]{type A = AB},
     u2: Unapply2[Bifunctor, AB]{type A = A0; type B = B0},
     l: Leibniz.===[AB, LazyEither[A0, B0]]
-  ): LazyEitherT[u1.M, A0, B0] = LazyEitherT(l.subst[u1.M](u1(fab)))
+  ): LazyEitherT[u1.M, u2.A, u2.B] = LazyEitherT(l.subst[u1.M](u1(fab)))
```

Comment:
Ehh.....this is tricky in the face of expected-type inference (the former propagates while the latter doesn't), best leave it as it was.